storage/innobase/xtrabackup/src/backup_mysql.cc

```diff
@@ -407,6 +407,23 @@ static bool check_server_version(unsigned long version_number,
     }
   }
 
+  std::string mysql_version = MYSQL_SERVER_VERSION;
+
+  // replace 8.0.21 first . with 0 and second . with null
+  std::size_t found = mysql_version.find_first_of(".");
+  if (found != std::string::npos) mysql_version[found] = '0';
+  found = mysql_version.find_first_of(".");
+  if (found != std::string::npos)
+    mysql_version.erase(mysql_version.begin() + found);
+
+  unsigned long mysql_version_int = stoi(mysql_version);
+  if (!opt_no_server_version_check && mysql_version_int < version_number) {
+    msg("Error: Unspported server version %s, Use latest version of PXB "
+        "or to continue with risk, use --no-server-version-check",
+        MYSQL_SERVER_VERSION);
```

Comment:
> Error: Unspported server version %s

This is reporting PXB version (`MYSQL_SERVER_VERSION`), while it should be reporting MySQL server version - `version_number`.